### PR TITLE
Parse headline title as text, update transform and render

### PR DESCRIPTION
**Pull request by schoettl**
- [x] Parse headline title as text
- [x] First, merge #40 and #42
- [x] Extract tags from headline in transform step
- [x] Join consecutive `:text-normal` in text
- [x] Join consecutive `:text-normal` in headline
- [x] Write render function for `:text`
- [x] Update README samples

### CODE REVIEW

File: src/org_parser/render.cljc [hunk 1]
```diff
@@ -18,10 +20,17 @@
 ;; to be extenden to the full feature scope.
 
 
+(defn- serialize-text
```
Comment (schoettl) [inline src/org_parser/render.cljc]: @munen How does these multi-arity functions work with this keyworded lists? The input for this function can be something like `[:text [:text-normal "uiea"] [:text-bold "foo"]]`

File: src/org_parser/transform.cljc [hunk 2]
```diff
@@ -45,17 +46,31 @@
                    first
                    count)
         title (->> properties
-                   (property :title)
-                   (str/join " "))]
+                   (property :text))
+        tags (->>  []) ;; TODO extract tags from the last element of :text if it's a :text-normal; empty otherwise;
```
Comment (schoettl) [inline src/org_parser/transform.cljc]: This line actually does not only extract tags, it also transforms the last `:text-normal` list element. Therefore, write a helper function which returns both, title and tags.

File: src/org_parser/transform.cljc [hunk 3]
```diff
@@ -45,17 +46,31 @@
...
+                                               ;;:tags  tags
+                                               ;; TODO much more to come, e.g. planning info (already parsed)
+                                               }})))
 
 
 ;; content-line needs to simply drop the keyword
 (defmethod reducer :content-line [state [_ ast] raw]
   (reducer state ast raw))
 
+;; Merge consecutive :text-normal inside a :text list. They come from
```
Comment (schoettl) [inline src/org_parser/transform.cljc]: @munen How to do this transformation:
`[:text [:text-normal "uiea"] [:text-normal "/foo"]]` => `[:text [:text-normal "uiea/foo"]]`
I laid out an fold algorithm but cannot implement it ATM.

File: src/org_parser/transform.cljc [hunk 4]
```diff
@@ -36,7 +43,20 @@
...
+       vec))
+
+(defn- extract-tags [[_ s]]
+  "Given a [:text-normal 'xxx'], return the text-normal without tags
+  and the tags as a list."
+  (let [[tags & _] (re-find #"\s+(:[a-zA-Z0-9_@#%]+)+:\s*$" s)] ;; find tags by regex
+    (if (nil? tags)
+      [[:text-normal s] []]
+      [[:text-normal (subs s (- (count s) (count tags)))]
+       (->> tags str/trim #(str/split % #":") (filter #(not (= % ""))))])))
```
Comment (schoettl) [inline src/org_parser/transform.cljc]: @munen in the next two lines I explain what this should do.

The tags must be extracted here, not in EBNF for practical reasons, see EBNF.

**Comment by schoettl** [comment]
Hi @branch14 @munen 

I'd like to merge this soon to be able to continue with other pending branches.

There is a lot of room for tests (transformation and rendering) but I'd rather starting to introduce more full-roundtrip tests.

We made a pretty big step with this branch (see updates in README).

Also, I refactored and introduced `insta/transform`. I think it makes a lot transformations much simpler and less verbose because this function handles the buttom-up tree walk and update/replace for us.

**Review by munen — APPROVED** [review]
Hi @schoettl 

This is looking very good to me - big steps taken, indeed!

Feel free to merge this PR at your convenience.

**Pull request merged**